src/curses.rs

```diff
@@ -479,90 +413,116 @@ where
     }
 }
 
-/// Resets the curses `cur_term` TERMINAL pointer. Subsequent calls to [`curses::term()`](term())
-/// will return `None`.
-pub fn reset() {
-    let mut term = TERM.lock().expect("Mutex poisoned!");
-    if term.is_some() {
-        unsafe {
-            // Ignore the result of del_curterm() as the only documented error is that
-            // `cur_term` was already null.
-            let _ = sys::del_curterm(get_curterm());
-            let _ = sys::set_curterm(core::ptr::null());
-        }
-        *term = None;
-    }
+pub fn setup_fallback_term() -> Arc<Term> {
+    let mut global_term = TERM.lock().expect("Mutex poisoned!");
+    // These values extracted from xterm-256color from ncurses 6.4
```

Comment:
Note that we [already](https://github.com/fish-shell/fish-shell/pull/9026) try xterm-256color if $TERM could not be found.

What this changes is:

1. If TERM is xterm-256color and could not be found, we use our fallback xterm-256color, without an error message or trying the other fallbacks
2. If we can't find *any* of of $TERM or our fallback TERM values, we use xterm-256color. Before, we would go on without *any* terminfo, and a lot of things would just *break*.

Both of these would be a good idea even if we kept using curses.

They are, however, more helpful with the rest of this, because it is now a bit more likely that we don't get a terminfo database (because you need to install curses to build against it, and because curses would also read the hashed database or awkward locations for that database).